addon/src/setup-application-context.ts

```diff
@@ -88,29 +57,19 @@ export function setupRouterSettlednessTracking() {
   HAS_SETUP_ROUTER.set(context, true);
 
   const { owner } = context;
-  let router: Router | RouterService;
-  if (CAN_USE_ROUTER_EVENTS) {
-    // SAFETY: unfortunately we cannot `assert` here at present because the
-    // class is not exported, only the type, since it is not designed to be
-    // sub-classed. The most we can do at present is assert that it at least
-    // *exists* and assume that if it does exist, it is bound correctly.
-    const routerService = owner.lookup('service:router');
-    assert('router service is not set up correctly', !!routerService);
-    router = routerService as RouterService;
-
-    // track pending transitions via the public routeWillChange / routeDidChange APIs
-    // routeWillChange can fire many times and is only useful to know when we have _started_
-    // transitioning, we can then use routeDidChange to signal that the transition has settled
-    router.on('routeWillChange', () => (routerTransitionsPending = true));
-    router.on('routeDidChange', () => (routerTransitionsPending = false));
-  } else {
-    // SAFETY: similarly, this cast cannot be made safer because on the versions
-    // where we fall into this path, this is *also* not an exported class.
-    const mainRouter = owner.lookup('router:main');
-    assert('router:main is not available', !!mainRouter);
-    router = mainRouter as Router;
-    ROUTER.set(context, router);
-  }
+
+  // SAFETY: unfortunately we cannot `assert` here at present because the
+  // class is not exported, only the type, since it is not designed to be
+  // sub-classed. The most we can do at present is assert that it at least
+  // *exists* and assume that if it does exist, it is bound correctly.
+  const router = owner.lookup('service:router') as RouterService;
+  assert('router service is not set up correctly', !!router);
```

Comment:
..and this hunk, should be the only non-trivial modification (even then this is still pretty trivial, just following the lead of the linter and consolidated some variables)